### ISSUE TITLE
tests failing because of pathlib.PosixPath types in pickled objects

### ISSUE THREAD

**Issue by kafitzgerald** [NCAR/wrf_hydro_py#169]
test_simulation_pickle and test_simulation_sub_obj_pickle are currently failing because deepdiff can't compare the pathlib.PosixPath types within these objects.  

Adding  exclude_types={pathlib.PosixPath} as an argument to DeepDiff will make the tests pass, but may circumvent intended testing.

**Comment by jmccreight** [comment]
I wonder if changing them to pathlib.Path objects would do the trick as well.

**Comment by jmccreight** [comment]
pathlib.Path does not seem to fix.

**Comment by kafitzgerald** [comment]
Still sorting out details, but just documenting here that the tests fail starting with version 4 of deepdiff thus the fix w/ requiring v3.3.0.

**Comment by jmccreight** [comment]
ha: https://github.com/seperman/deepdiff/issues/103

**Comment by jmccreight** [comment]
I assume i subclassed this to avoid waiting forever for this PR, as in the second part: 
https://github.com/seperman/deepdiff/pull/104

Maybe that's what's breaking.... something changes WRT this subclass:
https://github.com/NCAR/wrf_hydro_py/blob/d69f33c574a0453979681d8e30a476558d2697d9/wrfhydropy/core/ensemble_tools.py#L40

**Comment by jmccreight** [comment]
Fun with subclassing

**Comment by kafitzgerald** [comment]
Yeah, I think the updates to DeepDiff for v4 now trigger an exception which results in "unprocessed" output before your new function in the subclass deals with the pathlib.PosixPath stuff.  I started trying to edit the subclass (unless we want to ignore these, that seems like the way to go), but ran out of time for now.
